akka-actor/src/main/scala/akka/dispatch/affinity/ImmutableIntMap.scala

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.remote.artery
+
+import scala.annotation.tailrec
+import akka.util.OptionVal
+import scala.reflect.ClassTag
+import java.util.Arrays
+import akka.util.HashCode
+
+/**
+ * INTERNAL API
+ */
+private[akka] object ImmutableIntMap {
+  def empty: ImmutableIntMap =
+    new ImmutableIntMap(Array.emptyIntArray, Array.empty)
+
+  private val MaxScanLength = 10
+}
+
+/**
+ * INTERNAL API
+ * Specialized Map for primitive `Int` keys to avoid allocations (boxing).
+ * Keys and values are backed by arrays and lookup is performed with binary
+ * search. It's intended for rather small (<1000) maps.
+ */
+private[akka] class ImmutableIntMap private (
```

Comment:
It's up to you if you want to investigate.
Note that there is already `immutable.IntMap` for the case when the value is not a primitive, so I'm not sure we need anything more generic?